src/System.Net.NameResolution/tests/PalTests/NameResolutionPalTests.cs

```diff
@@ -164,14 +164,22 @@ public void TryGetAddrInfo_HostName_TryGetNameInfo()
             IPHostEntry hostEntry;
             int nativeErrorCode;
             SocketError error = NameResolutionPal.TryGetAddrInfo(hostName, out hostEntry, out nativeErrorCode);
+            if (error == SocketError.HostNotFound)
+            {
+                // On Unix, getaddrinfo returns host not found, if all the machine discovery settings on the local network
+                // is turned off. Hence dns lookup for it's own hostname fails.
```

Comment:
Is it possible to query somehow to see whether that's the cause?  If we had a bug and TryGetAddrInfo, for example, always returned HostNotFound for things other than "localhost", it seems like we'd sweep it under the rug.  Is there another host name we could check for, besides the current host name?  Could we, for example, lookup microsoft.com or bing.com or something like that?  It could be in a different test, if this one is specific to HostName.